src/main/java/com/pinterest/secor/util/orc/VectorColumnFiller.java

```diff
@@ -166,13 +165,28 @@ public void convert(JsonElement value, ColumnVector vect, int row) {
     }
 
     static class MapColumnConverter implements JsonConverter {
-        private JsonConverter[] childrenConverters;
+        private JsonConverter[] childConverters;
 
         public MapColumnConverter(TypeDescription schema) {
-            List<TypeDescription> kids = schema.getChildren();
-            childrenConverters = new JsonConverter[kids.size()];
-            for (int c = 0; c < childrenConverters.length; ++c) {
-                childrenConverters[c] = createConverter(kids.get(c));
+            assertKeyType(schema);
+
+            List<TypeDescription> childTypes = schema.getChildren();
+            childConverters = new JsonConverter[childTypes.size()];
+            for (int c = 0; c < childConverters.length; ++c) {
+                childConverters[c] = createConverter(childTypes.get(c));
+            }
+        }
+
+        /**
+         * Rejects non-string keys. This is a limitation imposed by JSON specifications that only allows strings
+         * as keys.
+         */
+        private void assertKeyType(TypeDescription schema) {
+            TypeDescription keyType = schema.getChildren().get(0);
```

Comment:
Does it guarantee that schema has Children?  And what are each children schema representing?

Reply:
Suppose we have an ORC schema like `map<string,int>` then the first child is a `TypeDescription` instance representing `string` type. Not sure if `schema` is guaranteed to have at least one child. We might want to test with malformed ORC schema like `map<>`. It may or may not reach this point when such schema is given.

Reply:
Also make sure the passed-in 'schema' argument really represents the map schema

Reply:
> Also make sure the passed-in 'schema' argument really represents the map schema

It certainly wouldn't hurt to do so, but I feel like this is somewhat redundant as it is already ensured by `VectorColumnFilter.createConverter()`. Its very logic doesn't leave any room for other types to fall into `MapColumnConverter`.

```
    public static JsonConverter createConverter(TypeDescription schema) {
        switch (schema.getCategory()) {

        // ...

        case MAP:
            return new MapColumnConverter(schema);
        default:
            throw new IllegalArgumentException("Unhandled type " + schema);
        }
    }
```

If you still insist doing so, I would suggest to add some common type checking logic in `JsonConverter` interface or in `VectorColumnFiller` class so that other types could also benefit from it. Any thoughts on this?